### PR TITLE
Support set_next_input

### PR DESCRIPTION
**Pull request by IanMatthewHuff**
For #4566 

-   [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [x] Title summarizes what is changing.
-   [x] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [x] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [x] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

File: src/client/datascience/notebook/helpers/executionHelpers.ts [hunk 1]
```diff
@@ -97,6 +98,34 @@ export async function updateCellWithErrorStatus(
...
+// Update the code contents of the cell
+export async function updateCellCode(cell: NotebookCell, text: string) {
+    // Use Workspace edit to apply a replace to the full cell text
+    const edit = new WorkspaceEdit();
+    edit.replace(
+        cell.document.uri,
+        new Range(cell.document.lineAt(0).range.start, cell.document.lineAt(cell.document.lineCount - 1).range.end),
+        text
+    );
+    await workspace.applyEdit(edit);
```
Comment (IanMatthewHuff) [inline src/client/datascience/notebook/helpers/executionHelpers.ts:110]: Quick note here. We used to guard against direct access to vscode classes using our injected interfaces. But with native notebooks and our .vscode tests this doesn't seem as important to me. I think that there is access to workspace edit off of IDocumentManager so I could use that, but to me it doesn't seem needed here.

Comment (IanMatthewHuff) [reply src/client/datascience/notebook/helpers/executionHelpers.ts:110]: But open to the idea if we think we should still do this.

Comment (DonJayamanne) [reply src/client/datascience/notebook/helpers/executionHelpers.ts:110]: I think this is perfectly fine, fewer interfaces/dependencies injected makes the code easier to manage, at least IMHO.

File: src/test/datascience/notebook/executionService.vscode.test.ts [hunk 2]
```diff
@@ -927,6 +927,41 @@ suite('DataScience - VSCode Notebook - (Execution) (slow)', function () {
...
+
+        // Check our output, one cell should have been inserted, and one been replaced
+        expect(cellsPostExecute.length).to.equal(3);
+        expect(cellsPostExecute[0].document.getText()).to.equal(
+            dedent`
+            import IPython
+            IPython.get_ipython().set_next_input("print('INSERT')")`
+        );
+        expect(cellsPostExecute[1].document.getText()).to.equal("print('INSERT')");
+        expect(cellsPostExecute[2].document.getText()).to.equal("print('REPLACE')");
```
Comment (DonJayamanne) [inline src/test/datascience/notebook/executionService.vscode.test.ts:962]: Awesome.......nice to have end to end tests, fewer items to manually test during bug bashes & fewer items yet for CTI...

**Review by DonJayamanne — APPROVED** [review]

File: src/test/datascience/notebook/executionService.vscode.test.ts (hunk 2 above)
Comment (IanMatthewHuff) [reply src/test/datascience/notebook/executionService.vscode.test.ts:962]: @DonJayamanne Yeah, I had issues with getting the test to run due to Python extension / winston issues. But once it was working this test was super easy and straight forward to write. I like not having to maintain a layer of test code as well.

**Review by DavidKutu — APPROVED** [review]

**Review by rchiodo — APPROVED** [review]

**Comment by IanMatthewHuff** [comment]
Only failure is flakey switch kernels test (failing in other recent PRs) so pushing.

**Pull request merged**